pkg/services/authz/rbac/service.go

```diff
@@ -307,10 +309,18 @@ func (s *Service) checkInheritedPermissions(ctx context.Context, scopeMap map[st
 
 	ctxLogger := s.logger.FromContext(ctx)
 
-	folderMap, err := s.buildFolderTree(ctx, req.Namespace)
-	if err != nil {
-		ctxLogger.Error("could not build folder and dashboard tree", "error", err)
-		return false, err
+	var folderMap map[string]FolderNode
```

Comment:
Actually, could we check/set cache in `buildFolderTree`? I need to fetch folder tree for listing as well, so would be good to have one place where the cache is being written/read. But I'm also happy to make that change in my PR (https://github.com/grafana/grafana/pull/98220).